vm/src/obj/objstr.rs

```diff
@@ -837,6 +839,33 @@ impl PyString {
             false
         }
     }
+
+    #[pymethod]
+    fn encode(
+        &self,
+        encoding: OptionalArg<PyObjectRef>,
```

Comment:
use PyStringRef to avoid type checking for "encoding" and "errors" wich will always be string

Reply:
By trying it, I found I can't control the exception message with it. Do you have any idea about it?

Reply:
RustPython doesn't aim to have exactly the same error message as cpython. But it's better to let the type checking "automatic" in function type.

Reply:
I am not sure about error messages. If we can keep them the same or similar with small cost, why not? Is there priority about this? @windelbouwman

Reply:
The error messages to my opinion do not have to be similar to cpython. While I worked with rust, I find the error messages super useful in rust, maybe we could try something new in rustpython as well with regards to error messages. So in this case, I agree with @jgirardet to use the type checker with `PyStringRef` in this case.